astropy/timeseries/periodograms/lombscargle/implementations/cython_impl.pyx

```diff
@@ -58,6 +58,10 @@ def lombscargle_cython(t, y, dy, frequency, normalization='standard',
         dy = 1
 
     t, y, dy = np.broadcast_arrays(t, y, dy)
+
+    t.flags.writeable = True
```

Comment:
I'm not sure it is right to just make the arrays writable - if any values are broadcast, something like `y -= ...` could very easily do the wrong thing. Will check.